tests/test_output.py

```diff
@@ -0,0 +1,511 @@
+    output.pbar_finished.is_set.return_value = True
+
+    original_stdout = sys.stdout
+    captured_output = StringIO()
+    sys.stdout = captured_output
+
+    stats = "Analyzed IPs: 20"
+    output.handle_printing_stats(stats)
+
+    sys.stdout = original_stdout
+    assert captured_output.getvalue().strip() == stats
+
+
+@pytest.mark.parametrize(
+    "output_verbose, input_verbose, expected_result",
+    [  # Testcase1: Input verbose less than output verbose
+        (2, 1, True),
+        # Testcase2: Input verbose equal to output verbose
+        (2, 2, True),
+        # Testcase3: Input verbose greater than output verbose
+        (2, 3, False),
+        # Testcase4: Input verbose is 0
+        (1, 0, False),
+        # Testcase5: Input verbose is negative
+        (1, -1, False),
+        # Testcase6: Input verbose greater than the maximum
+        (3, 4, False),
+    ],
+)
+def test_enough_verbose(output_verbose, input_verbose, expected_result):
+    """Test that the enough_verbose method returns the correct result."""
+    output = ModuleFactory().create_output_obj()
+    output.verbose = output_verbose
+
+    assert output.enough_verbose(input_verbose) == expected_result
+
+
+@pytest.mark.parametrize(
+    "output_debug, input_debug, expected_result",
+    [  # Testcase1: Input debug less than output debug
+        (2, 1, True),
+        # Testcase2: Input debug equal to output debug
+        (2, 2, True),
+        # Testcase3: Input debug greater than output debug
+        (2, 3, False),
+        # Testcase4: Input debug is 0
+        (1, 0, False),
+        # Testcase5: Input debug is negative
+        (1, -1, False),
+        # Testcase6: Input debug greater than the maximum
+        (3, 4, False),
+    ],
+)
+def test_enough_debug(output_debug, input_debug, expected_result):
+    """Test that the enough_debug method returns the correct result."""
+    output = ModuleFactory().create_output_obj()
+    output.debug = output_debug
+
+    assert output.enough_debug(input_debug) == expected_result
+
+
+@patch("slips_files.core.output.Output.print")
+@patch("slips_files.core.output.Output.log_line")
+@patch("slips_files.core.output.Output.log_error")
+def test_output_line_all_outputs(mock_log_error, mock_log_line, mock_print):
+    output = ModuleFactory().create_output_obj()
+    output.verbose = 2
+    output.debug = 2
+
+    msg = {
+        "from": "SenderName",
+        "txt": "Normal message",
+        "verbose": 2,
+        "debug": 1,
+    }
+
+    output.output_line(msg)
+
+    mock_print.assert_called_with(msg["from"], msg["txt"])
+    mock_log_line.assert_called_with(msg)
+    mock_log_error.assert_called_with(msg)
+
+
+@patch("slips_files.core.output.Output.print")
+@patch("slips_files.core.output.Output.log_line")
+@patch("slips_files.core.output.Output.log_error")
+def test_output_line_no_outputs(mock_log_error, mock_log_line, mock_print):
+    output = ModuleFactory().create_output_obj()
```

Comment:
here we can add a docstring saying that it didnt print because the given verbose (3) is more than the module's verbose (2)

Reply:
because the function name is not enough to clarify this